python/pyspark/sql/pandas/serializers.py

```diff
@@ -155,7 +156,11 @@ def create_array(s, t):
             if t is not None and pa.types.is_timestamp(t):
                 s = _check_series_convert_timestamps_internal(s, self._timezone)
             try:
-                array = pa.Array.from_pandas(s, mask=mask, type=t, safe=self._safecheck)
+                if type(s.dtype) == CategoricalDtype:
```

Comment:
I think you can just use `pd.CategoricalDtype` and avoid the above import

Reply:
Done.